### PR TITLE
Add Steering Override

### PR DESCRIPTION
**Pull request by k-cleary**
Prior to this PR, we had no way of detecting steering operator overrides with the current model of allowing the full range of allowable outputs, not limiting anything except for where it would fault the vehicle. This PR applies a smoothing filter to the analogRead function of the steering module to smooth out noisy cases where the wheel hits the mechanical limits of the steering rack, allowing us to detect when the differential value is consistently higher or lower than the baseline, which indicates operator interference and triggers an override.

### CODE REVIEW

**Comment by k-cleary** [comment]
This has been updated in order to lower the steering override threshold and make it easier for a manual takeover. The update has been field tested.

File: firmware/steering/src/steering_control.cpp [hunk 1]
```diff
@@ -43,9 +44,21 @@ void check_for_operator_override( void )
 
         read_torque_sensor( &torque );
 
-        float diff = (float)torque.high - (float)torque.low;
+        uint16_t unfiltered_diff = abs((int)torque.high - (int)torque.low);
```
Comment (austinjmorlan) [inline firmware/steering/src/steering_control.cpp]: This section has a few style inconsistences.

Lines 47, 51, and 61 have parentheses with missing spaces inside of them.

Lines 57-59 should be indented once more.

Comment (k-cleary) [reply firmware/steering/src/steering_control.cpp]: Updated with fb5bb4e

File: firmware/brake/tests/features/step_definitions/receiving_messages.cpp [hunk 2]
```diff
@@ -11,15 +11,17 @@ GIVEN("^the left brake sensor reads (.*)$")
 {
     REGEX_PARAM(int, sensor_val);
 
-    g_mock_arduino_analog_read_return = sensor_val;
+    g_mock_arduino_analog_read_return[14] = sensor_val;
+    g_mock_arduino_analog_read_return[14] = sensor_val;
```
Comment (p6363091) [inline firmware/brake/tests/features/step_definitions/receiving_messages.cpp]: Is line 15 necessary? Does it do anything different compared to line 14?

File: firmware/brake/tests/features/step_definitions/receiving_messages.cpp [hunk 3]
```diff
@@ -11,15 +11,17 @@ GIVEN("^the left brake sensor reads (.*)$")
...
 }
 
 
 GIVEN("^the right brake sensor reads (.*)$")
 {
     REGEX_PARAM(int, sensor_val);
 
-    g_mock_arduino_analog_read_return = sensor_val;
+    g_mock_arduino_analog_read_return[13] = sensor_val;
+    g_mock_arduino_analog_read_return[13] = sensor_val;
```
Comment (p6363091) [inline firmware/brake/tests/features/step_definitions/receiving_messages.cpp]: Is line 24 necessary? Does it do anything different compared to line 23?

File: firmware/brake/tests/features/step_definitions/receiving_messages.cpp (hunk 2 above)
Comment (austinjmorlan) [reply firmware/brake/tests/features/step_definitions/receiving_messages.cpp]: 857217f

File: firmware/brake/tests/features/step_definitions/receiving_messages.cpp (hunk 3 above)
Comment (austinjmorlan) [reply firmware/brake/tests/features/step_definitions/receiving_messages.cpp]: 857217f

**Pull request merged**